brainiak/event_seg/event.py

```diff
@@ -0,0 +1,312 @@
+"""Event segmentation using a Hidden Markov Model"""
+
+# Authors: Chris Baldassano and Cătălin Iordan (Princeton University)
+
+import numpy as np
+from scipy import stats
+import logging
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "EventSegment",
+]
+
+
+class EventSegment:
+    """Class for learning and representing an event segmentation of
+    continuous fMRI data
+
+    Attributes
+    ----------
+    n_events: int
+        Number of segments to learn
+
+    step_var: lambda : default lambda step: 4 * (0.98 ** (step - 1))
+        The Gaussian variance to use during fitting, as a function of the
+        number of steps. Should decrease slowly over time.
+
+    max_steps: int : default 500
+        Maximum number of steps to run during fitting
+
+
+    p_start, p_end: row vectors
+        initial and final prior distributions over events
+
+    P: array
+        HMM transition matrix
+
+    ll: array
+        Log-likelihood for training datasets over the course of training
+
+    segments: list of (time by event) arrays
+        Learned (soft) segmentation for training datasets
+
+    event_var: float
+        Gaussian variance at the end of learning
+
+    event_pat: voxel by event array
+        Learned mean patterns for each event
+    """
+
+    def __init__(self, n_events,
+                 step_var=lambda step: 4 * (0.98 ** (step - 1)),
+                 max_steps=500):
+        self.n_events = n_events
+        self.step_var = step_var
+        self.max_steps = max_steps
+
+        # Set up transition matrix, with final sink state
+        self.p_start = np.zeros((1, n_events + 1))
+        self.p_start[0, 0] = 1
+        self.P = np.vstack((np.hstack((0.5 * np.diag(np.ones(n_events)) +
+                                       0.5 * np.diag(np.ones(n_events - 1), 1),
```

Comment:
Binary operators should be at the beginning of the line, not at the end:
https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator